### PR TITLE
fix: preserve OpenAI system messages as Claude top-level system

### PR DESCRIPTION
**Pull request by shellus**
## Summary
- map OpenAI chat/completions `role=system` messages to Claude top-level `system`
- stop degrading system prompts into the first synthetic `user` message
- add regression tests for single and multiple system message inputs

## Root Cause
The OpenAI -> Claude translator handled `role == "system"` by creating a synthetic `{"role":"user","content":[]}` message and appending the system text there. This lost Anthropic's top-level system semantics on the `chat/completions -> CLIProxyAPI -> Claude` path.

## Testing
- `proxy-env docker run --rm -e GOPROXY=https://goproxy.cn,direct -e GOSUMDB=sum.golang.google.cn -v /data/compose/cliproxy/source:/workspace -w /workspace golang:1.26-alpine sh -lc '/usr/local/go/bin/go test ./internal/translator/claude/openai/chat-completions -count=1'`

### CODE REVIEW

File: internal/translator/claude/openai/chat-completions/claude_openai_request_test.go [hunk 1]
```diff
@@ -135,3 +135,71 @@ func TestConvertOpenAIRequestToClaude_ToolResultURLImageOnly(t *testing.T) {
...
+		t.Fatalf("Expected first system text %q, got %q", "Rule 1", got)
+	}
+	if got := system[1].Get("text").String(); got != "Rule 2" {
+		t.Fatalf("Expected second system text %q, got %q", "Rule 2", got)
+	}
+
+	messages := resultJSON.Get("messages").Array()
+	if len(messages) != 1 {
+		t.Fatalf("Expected 1 non-system message, got %d. Messages: %s", len(messages), resultJSON.Get("messages").Raw)
+	}
```
[the remark this answers is not in the document]
Comment (shellus) [reply internal/translator/claude/openai/chat-completions/claude_openai_request_test.go:204]: Fixed. Added assertions in `TestConvertOpenAIRequestToClaude_MultipleSystemMessagesMergedIntoTopLevelSystem` to verify the remaining non-system message is the expected `user` message with `Hello` content.

File: internal/translator/claude/openai/chat-completions/claude_openai_request.go [hunk 2]
```diff
@@ -165,29 +165,22 @@ func ConvertOpenAIRequestToClaude(modelName string, inputRawJSON []byte, stream
...
-					systemMsg := []byte(`{"role":"user","content":[]}`)
-					out, _ = sjson.SetRawBytes(out, "messages.-1", systemMsg)
-					systemMessageIndex = messageIndex
-					messageIndex++
-				}
 				if contentResult.Exists() && contentResult.Type == gjson.String && contentResult.String() != "" {
 					textPart := []byte(`{"type":"text","text":""}`)
 					textPart, _ = sjson.SetBytes(textPart, "text", contentResult.String())
-					out, _ = sjson.SetRawBytes(out, fmt.Sprintf("messages.%d.content.-1", systemMessageIndex), textPart)
+					out, _ = sjson.SetRawBytes(out, "system.-1", textPart)
```
[the remark this answers is not in the document]
Comment (shellus) [reply internal/translator/claude/openai/chat-completions/claude_openai_request.go:177]: Fixed. The translator now adds a minimal fallback `user` turn only when the OpenAI input is system-only, so we preserve Claude top-level `system` semantics without duplicating the system prompt into a synthetic user message. Added a regression test for the system-only case as well.

**Comment by shellus** [comment]
Per `translator-path-guard`, I opened the required maintenance issue for this translator change: #2312\n\nIssue link: https://github.com/router-for-me/CLIProxyAPI/issues/2312\n\nThis PR remains available as an implementation reference for maintainers if they want to cherry-pick or reproduce the fix internally.

**Review by luispater — COMMENTED** [review]
## Summary
This PR correctly maps OpenAI `role=system` messages to Claude top-level `system` blocks, instead of degrading them into a synthetic `user` message. It also adds a safety fallback for system-only inputs and includes focused regression tests.

## Key findings
- Blocking: none.
- Non-blocking: test coverage is good for the core fix; edge-case coverage could be expanded for empty or unsupported system content and end-to-end acceptance of the fallback message shape.

## Test plan
- Reviewed the changed translator logic and new unit tests in:
  - `internal/translator/claude/openai/chat-completions/claude_openai_request.go`
  - `internal/translator/claude/openai/chat-completions/claude_openai_request_test.go`
- Checked PR CI status and ignored `translator-path-guard` as requested.
- `build` check is passing.

## Follow-ups
- Add a regression test for `system` messages with empty or unsupported content blocks to document expected drop behavior explicitly.
- Add an integration-style test that validates the `system`-only fallback message (`user` with empty text) is accepted end-to-end by the Claude execution path.

This is an automated Codex review result and still requires manual verification by a human reviewer.

**Review by luispater — APPROVED** [review]
